src/internal/packager2/publish_test.go

```diff
@@ -0,0 +1,397 @@
+		})
+	}
+}
+
+func TestPublishCopySHA(t *testing.T) {
+	// t.Parallel()
+
+	tt := []struct {
+		name             string
+		packageToPublish string
+		opts             PublishPackageOpts
+	}{
+		{
+			name:             "Publish package",
+			packageToPublish: "testdata/zarf-package-test-amd64-0.0.1.tar.zst",
+			opts: PublishPackageOpts{
+				WithPlainHTTP: true,
+				Architecture:  "amd64",
+				Concurrency:   3,
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// file write is happening during publish oras.PackManifest, we should replace defer remove that
+			ctx := testutil.TestContext(t)
+			registryRef := createRegistry(t, ctx)
+
+			// Publish test package
+			err := PublishPackage(ctx, tc.packageToPublish, registryRef, tc.opts)
+			require.NoError(t, err)
+
+			// Setup destination registry
+			dstRegistryRef := createRegistry(t, ctx)
+
+			// This gets the test package digest from the first package publish
+			localRepo := &remote.Repository{PlainHTTP: true}
+			ociSrc := fmt.Sprintf("%s/%s", registryRef.String(), "test:0.0.1")
+			localRepo.Reference, err = registry.ParseReference(ociSrc)
+			require.NoError(t, err)
+			indexDesc, err := oras.Resolve(ctx, localRepo, ociSrc, oras.ResolveOptions{})
+			require.NoError(t, err)
+			src := fmt.Sprintf("%s/%s@%s", registryRef.String(), "test:0.0.1", indexDesc.Digest)
+			srcRef, err := registry.ParseReference(src)
+			require.NoError(t, err)
+
+			dst := fmt.Sprintf("%s/%s", dstRegistryRef.String(), "test:0.0.1")
+			dstRef, err := registry.ParseReference(dst)
+			require.NoError(t, err)
+
+			// TODO: test case for sha on the src but not on dst
+			opts := PublishFromOCIOpts{
+				WithPlainHTTP: tc.opts.WithPlainHTTP,
+				Architecture:  tc.opts.Architecture,
+				Concurrency:   tc.opts.Concurrency,
+			}
+
+			// Publish test package to the destination registry
+			err = PublishFromOCI(ctx, srcRef, dstRef, opts)
+			require.NoError(t, err)
+
+			// We want to pull the package and sure the content is the same as the local package
+			layoutExpected, err := layout2.LoadFromTar(ctx, tc.packageToPublish, layout2.PackageLayoutOptions{})
+			require.NoError(t, err)
+			// Publish creates a local oci manifest file using the package name, delete this to clean up test name
+			defer os.Remove(layoutExpected.Pkg.Metadata.Name)
+			// Format url and instantiate remote
+			packageRef, err := zoci.ReferenceFromMetadata(dstRegistryRef.String(), &layoutExpected.Pkg.Metadata, &layoutExpected.Pkg.Build)
+			require.NoError(t, err)
+
+			pkgRefsha := fmt.Sprintf("%s@%s", packageRef, indexDesc.Digest)
+
+			layoutActual := pullFromRemote(t, ctx, pkgRefsha, "amd64")
+			require.Equal(t, layoutExpected.Pkg, layoutActual.Pkg, "Uploaded package is not identical to downloaded package")
+		})
+	}
+}
+
+func TestPublishCopyTag(t *testing.T) {
+	// FIXME(mkcp): TestPublishCopy panics when running in parallel on test count 2 or higher.
+	// t.Parallel()
```

Comment:
I double checked this one locally, and the unit tests ran w/o panic using `t.Parallel()` and `test.count={2,5,10,100}`.  Might be some other resource collision that changed since we wrote this. IMO, delete both the FIXME and the commented out // t.Parallel()